include/ginkgo/core/log/logger.hpp

```diff
@@ -658,10 +691,40 @@ class EnableLogging : public PolymorphicBase {
 
     void clear_loggers() override { loggers_.clear(); }
 
+private:
+    template <typename OtherLoggable, typename OtherBase>
+    friend class EnableLogging;
+
+    template <size_type Event, typename ConcreteLoggableT, typename = void>
+    struct propagate_log_helper {
```

Comment:
IMO for profiling an application, neither operations nor high-level LinOp events are sufficient to understand the performance characteristics. Operations catch non-apply/copy function calls, apply gives a nested hierarchy of solver components, both of which I would consider useful. If we tell users "you need to attach this logger to all your LinOps", I would assume they would avoid using it for more complex applications, where it could really shine otherwise.

Reply:
What do you think about the new approach?
gko::log::propagation_mode controls whether an Executor will receive high-level events

Reply:
If user wants the detail/the all performance of application, they can simply apply the logger to executor.
If they want more specific range about some operation, they apply it to the operation.
I think propagation_mode is good for both.
If we propagate the logger to executor, they just know what's the activity from ginkgo.
the future usage naming annotation from user should not propagate to ginkgo.